persist/fs/commitlog/reader.go

```diff
@@ -165,16 +283,10 @@ func (r *reader) readChunk() ([]byte, error) {
 		return nil, err
 	}
 
-	// Extend buffer as necessary
-	if len(r.dataBuffer) < int(size) {
-		diff := int(size) - len(r.dataBuffer)
-		for i := 0; i < diff; i++ {
-			r.dataBuffer = append(r.dataBuffer, 0)
-		}
-	}
+	newBuf := make([]byte, size, size)
```

Comment:
Yeah I actually had a mental TODO to ask you about that. I've never done object pooling before. Do I just do a .Get() when I need a new one and a .Put() when I'm done with it?